### ISSUE TITLE
Video freeze with 7.0.6.2

### ISSUE THREAD

**Issue by Shadowevil1666** [anxdpanic/plugin.video.youtube#737]
### Context

Please provide any relevant information about your setup

* Add-on Version: 7.0.6.2
* Kodi Version: 21
* Kodi GUI Language: german
* Operating System: Coreelec-ng 21 latest nightly
* Operating language:German 

I have tried 4 Vids, every Video freeze after 2 Minutes. Audio runs again after few seconds. But Video still freeze. I go back to 7.0.6.1 with no Problem 

------

### Expected Behavior

Please describe the behavior you are expecting.

------

### Current Behavior

What is the current behavior?

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. 
2. 
3. 

------

### Log

Please include a complete [debug log](https://kodi.wiki/view/Log_file).

------

### Additional Information

Please provide any additional information that may be helpful.

------

**Comment by xalaros** [comment]
On Android with Kodi 21 and YouTube v7.0.6.2 can't appear embedded subtitles although I have enabled them on settings.
Also sometimes videos playing 4Κ and othertimes the same ones playing 720p.

**Comment by neo-neo1** [comment]
You're going to be asked to post a debug log.

And @xalaros you should start a separate post – this is unrelated. And post a debug log.

**Comment by MoojMidge** [comment]
> You're going to be asked to post a debug log.
> 
> And @xalaros you should start a separate post – this is unrelated. And post a debug log.

Lol

While I may sound like a broken record, I actually do try and reproduce all these issues, even ones submitted without a log. Where a log is not provided I will usually take extra time to re-review all the code that may possibly effect the issue reported to try and figure out what may be causing it.

This is a massive waste of time and something that I will no longer do, as it means that I am spending more time trying to figure out information that people who are raising these issues already have, rather than actually working on the plugin.

**Comment by Tomte-Fan** [comment]
I'm seeing the same issues, here is a debug log showing the problem.

First playback it just stopped playing, second playback only gave me the option of 720p as max and third playback the image hung and audio continued for a short bit and then it was just noise and I stopped it.
[kodi.log](https://github.com/anxdpanic/plugin.video.youtube/files/15188373/kodi.log)

**Comment by damnms** [comment]
just thought i report that bug (playback freezes after a few seconds/minutes), but this debug log from tomte looks like my errors. i get ton of those:
```
2024-05-02 17:34:28.737 T:4537     info <general>: CAESinkALSA::InitializeHW - Your hardware does not support AE_FMT_FLOAT, trying other formats
2024-05-02 17:34:28.737 T:4537     info <general>: CAESinkALSA::InitializeHW - Using data format AE_FMT_S24NE4
2024-05-02 17:34:28.756 T:4506     info <general>: Constructing CRendererAML
2024-05-02 17:34:58.721 T:8636     info <general>: Skipped 1 duplicate messages..
2024-05-02 17:34:58.721 T:8636     info <general>: ProcessDecoderOutput: Changed max allowed Out-Of-Sync value to 41 ms due self-learning
2024-05-02 17:34:59.640 T:8632    error <general>: AddOnLog: inputstream.adaptive: [AS-0] Download failed, HTTP error 403: https://rr2---sn-1gi7znek.googlevideo.com/videoplayback?expire=1714685667&ei=g7IzZp-rBtKx6dsP-L2HmAY&ip=85.195.252.80&id=o-ANhHRYXW5C9tqwGi7otCYH3hwv1R4Nn7DA1Xu7Piw1fC&itag=303&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=-d&mm=31%2C29&mn=sn-1gi7znek%2Csn-1gieen7e&ms=au%2Crdu&mv=m&mvi=2&pl=21&initcwndbps=1807500&spc=UWF9f_nld6fuWr6E3-A3oUowCn5EOdCBisyN-dMDmH0-2nINHQ&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1154393145&dur=2728.133&lmt=1714516450532507&mt=1714663749&fvip=4&keepalive=yes&c=ANDROID&txp=4437434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAORTqCz0CFKlk73k28m94TIAkBTkO9f2lNGbJOhCIZ7eAiEAu6jGMGViPLAtGVdcoqEQSIeh3K3ZFadAsSitF6xFvVo%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHWaYeowRgIhANfFOkwfVa6vHQof6orFaSeJApDzAXvnGGS9ji52Yh0wAiEAiesILEeWcV9BfwYziI62HFBPwRBbyuCOpeifzhA1eu4%3D&range=0-1154393145
2024-05-02 17:35:00.141 T:8632  warning <general>: AddOnLog: inputstream.adaptive: [AS-0] Segment download failed, attempt 2...
2024-05-02 17:35:00.156 T:8632    error <general>: AddOnLog: inputstream.adaptive: [AS-0] Download failed, HTTP error 403: https://rr2---sn-1gi7znek.googlevideo.com/videoplayback?expire=1714685667&ei=g7IzZp-rBtKx6dsP-L2HmAY&ip=85.195.252.80&id=o-ANhHRYXW5C9tqwGi7otCYH3hwv1R4Nn7DA1Xu7Piw1fC&itag=303&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=-d&mm=31%2C29&mn=sn-1gi7znek%2Csn-1gieen7e&ms=au%2Crdu&mv=m&mvi=2&pl=21&initcwndbps=1807500&spc=UWF9f_nld6fuWr6E3-A3oUowCn5EOdCBisyN-dMDmH0-2nINHQ&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1154393145&dur=2728.133&lmt=1714516450532507&mt=1714663749&fvip=4&keepalive=yes&c=ANDROID&txp=4437434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAORTqCz0CFKlk73k28m94TIAkBTkO9f2lNGbJOhCIZ7eAiEAu6jGMGViPLAtGVdcoqEQSIeh3K3ZFadAsSitF6xFvVo%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHWaYeowRgIhANfFOkwfVa6vHQof6orFaSeJApDzAXvnGGS9ji52Yh0wAiEAiesILEeWcV9BfwYziI62HFBPwRBbyuCOpeifzhA1eu4%3D&range=0-1154393145
2024-05-02 17:35:00.657 T:8632  warning <general>: AddOnLog: inputstream.adaptive: [AS-0] Segment download failed, attempt 3...
2024-05-02 17:35:00.685 T:8632    error <general>: AddOnLog: inputstream.adaptive: [AS-0] Download failed, HTTP error 403: https://rr2---sn-1gi7znek.googlevideo.com/videoplayback?expire=1714685667&ei=g7IzZp-rBtKx6dsP-L2HmAY&ip=85.195.252.80&id=o-ANhHRYXW5C9tqwGi7otCYH3hwv1R4Nn7DA1Xu7Piw1fC&itag=303&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=-d&mm=31%2C29&mn=sn-1gi7znek%2Csn-1gieen7e&ms=au%2Crdu&mv=m&mvi=2&pl=21&initcwndbps=1807500&spc=UWF9f_nld6fuWr6E3-A3oUowCn5EOdCBisyN-dMDmH0-2nINHQ&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1154393145&dur=2728.133&lmt=1714516450532507&mt=1714663749&fvip=4&keepalive=yes&c=ANDROID&txp=4437434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAORTqCz0CFKlk73k28m94TIAkBTkO9f2lNGbJOhCIZ7eAiEAu6jGMGViPLAtGVdcoqEQSIeh3K3ZFadAsSitF6xFvVo%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHWaYeowRgIhANfFOkwfVa6vHQof6orFaSeJApDzAXvnGGS9ji52Yh0wAiEAiesILEeWcV9BfwYziI62HFBPwRBbyuCOpeifzhA1eu4%3D&range=0-1154393145

```
i really like the plugin but i somehow have the feeling that its becoming more buggy with every new release :/

**Comment by Kellett** [comment]
Another log for you. Videos never seem to play for more than a minute or two before hanging.

[kodi.log](https://github.com/anxdpanic/plugin.video.youtube/files/15191353/kodi.log)

**Comment by neo-neo1** [comment]
> just thought i report that bug (playback freezes after a few seconds/minutes), but this debug log from tomte looks like my errors. i get ton of those:
> 
> ```
> 2024-05-02 17:34:28.737 T:4537     info <general>: CAESinkALSA::InitializeHW - Your hardware does not support AE_FMT_FLOAT, trying other formats
> 2024-05-02 17:34:28.737 T:4537     info <general>: CAESinkALSA::InitializeHW - Using data format AE_FMT_S24NE4
> 2024-05-02 17:34:28.756 T:4506     info <general>: Constructing CRendererAML
> 2024-05-02 17:34:58.721 T:8636     info <general>: Skipped 1 duplicate messages..
> 2024-05-02 17:34:58.721 T:8636     info <general>: ProcessDecoderOutput: Changed max allowed Out-Of-Sync value to 41 ms due self-learning
> 2024-05-02 17:34:59.640 T:8632    error <general>: AddOnLog: inputstream.adaptive: [AS-0] Download failed, HTTP error 403: https://rr2---sn-1gi7znek.googlevideo.com/videoplayback?expire=1714685667&ei=g7IzZp-rBtKx6dsP-L2HmAY&ip=85.195.252.80&id=o-ANhHRYXW5C9tqwGi7otCYH3hwv1R4Nn7DA1Xu7Piw1fC&itag=303&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=-d&mm=31%2C29&mn=sn-1gi7znek%2Csn-1gieen7e&ms=au%2Crdu&mv=m&mvi=2&pl=21&initcwndbps=1807500&spc=UWF9f_nld6fuWr6E3-A3oUowCn5EOdCBisyN-dMDmH0-2nINHQ&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1154393145&dur=2728.133&lmt=1714516450532507&mt=1714663749&fvip=4&keepalive=yes&c=ANDROID&txp=4437434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAORTqCz0CFKlk73k28m94TIAkBTkO9f2lNGbJOhCIZ7eAiEAu6jGMGViPLAtGVdcoqEQSIeh3K3ZFadAsSitF6xFvVo%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHWaYeowRgIhANfFOkwfVa6vHQof6orFaSeJApDzAXvnGGS9ji52Yh0wAiEAiesILEeWcV9BfwYziI62HFBPwRBbyuCOpeifzhA1eu4%3D&range=0-1154393145
> 2024-05-02 17:35:00.141 T:8632  warning <general>: AddOnLog: inputstream.adaptive: [AS-0] Segment download failed, attempt 2...
> 2024-05-02 17:35:00.156 T:8632    error <general>: AddOnLog: inputstream.adaptive: [AS-0] Download failed, HTTP error 403: https://rr2---sn-1gi7znek.googlevideo.com/videoplayback?expire=1714685667&ei=g7IzZp-rBtKx6dsP-L2HmAY&ip=85.195.252.80&id=o-ANhHRYXW5C9tqwGi7otCYH3hwv1R4Nn7DA1Xu7Piw1fC&itag=303&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=-d&mm=31%2C29&mn=sn-1gi7znek%2Csn-1gieen7e&ms=au%2Crdu&mv=m&mvi=2&pl=21&initcwndbps=1807500&spc=UWF9f_nld6fuWr6E3-A3oUowCn5EOdCBisyN-dMDmH0-2nINHQ&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1154393145&dur=2728.133&lmt=1714516450532507&mt=1714663749&fvip=4&keepalive=yes&c=ANDROID&txp=4437434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAORTqCz0CFKlk73k28m94TIAkBTkO9f2lNGbJOhCIZ7eAiEAu6jGMGViPLAtGVdcoqEQSIeh3K3ZFadAsSitF6xFvVo%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHWaYeowRgIhANfFOkwfVa6vHQof6orFaSeJApDzAXvnGGS9ji52Yh0wAiEAiesILEeWcV9BfwYziI62HFBPwRBbyuCOpeifzhA1eu4%3D&range=0-1154393145
> 2024-05-02 17:35:00.657 T:8632  warning <general>: AddOnLog: inputstream.adaptive: [AS-0] Segment download failed, attempt 3...
> 2024-05-02 17:35:00.685 T:8632    error <general>: AddOnLog: inputstream.adaptive: [AS-0] Download failed, HTTP error 403: https://rr2---sn-1gi7znek.googlevideo.com/videoplayback?expire=1714685667&ei=g7IzZp-rBtKx6dsP-L2HmAY&ip=85.195.252.80&id=o-ANhHRYXW5C9tqwGi7otCYH3hwv1R4Nn7DA1Xu7Piw1fC&itag=303&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=-d&mm=31%2C29&mn=sn-1gi7znek%2Csn-1gieen7e&ms=au%2Crdu&mv=m&mvi=2&pl=21&initcwndbps=1807500&spc=UWF9f_nld6fuWr6E3-A3oUowCn5EOdCBisyN-dMDmH0-2nINHQ&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1154393145&dur=2728.133&lmt=1714516450532507&mt=1714663749&fvip=4&keepalive=yes&c=ANDROID&txp=4437434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAORTqCz0CFKlk73k28m94TIAkBTkO9f2lNGbJOhCIZ7eAiEAu6jGMGViPLAtGVdcoqEQSIeh3K3ZFadAsSitF6xFvVo%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHWaYeowRgIhANfFOkwfVa6vHQof6orFaSeJApDzAXvnGGS9ji52Yh0wAiEAiesILEeWcV9BfwYziI62HFBPwRBbyuCOpeifzhA1eu4%3D&range=0-1154393145
> ```
> 
> i really like the plugin but i somehow have the feeling that its becoming more buggy with every new release :/

That looks like a general network error – which more and more people have started to report as of late.

Unfortunately I believe this is the cause; [YouTube’s ad blocker crackdown now includes third-party apps](https://www.theverge.com/2024/4/15/24131338/youtube-ad-blocker-crackdown-mobile-apps)

"YouTube says its policies don’t allow “third-party apps to turn off ads because that prevents the creator from being rewarded for viewership.” The block targets third-party apps that use YouTube’s API to get videos interruption-free."

**Comment by MoojMidge** [comment]
Yeah part of the same thing I think. Only made the change a few days ago and has been working fine for me and others, but the checks they are doing must not be universally being applied.

Will revert back to not using the default client.

**Comment by damnms** [comment]
> Unfortunately I believe this is the cause

You mean, the adblock of this youtube plugin leads to that error? because from my workstation with ublock, all works perfectly fine. In the youtube plugin, the video starts but after a few minutes it freezes

**Comment by MoojMidge** [comment]
Its related to Youtube on Android

Anyway will be "fixed" in v7.0.6.3
